src/CMakeLists.txt

```diff
@@ -169,164 +157,45 @@ SET (HDRS
 	XMLParser.h
 )
 
-file(WRITE "${CMAKE_BINARY_DIR}/include/Globals.h"
-	"/* This file allows Globals.h to be included with an absolute path */\n#include \"${PROJECT_SOURCE_DIR}/Globals.h\"\n")
+set(FOLDERS
+	OSSupport HTTP Items Blocks Protocol Generating mbedTLS++ Bindings
+	WorldStorage Mobs Entities Simulator Simulator/IncrementalRedstoneSimulator
+	BlockEntities UI Noise
+)
+
+# Add all child source directories:
+foreach(folder ${FOLDERS})
+	add_subdirectory(${folder})
+endforeach(folder)
 
-include_directories("${CMAKE_BINARY_DIR}/include")
-include_directories (SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/../lib/TCLAP/include")
+# TODO: CMake includes PCH in every source file
+# Remove when every build has PCH enabled:
+file(WRITE "${CMAKE_BINARY_DIR}/include/Globals.h"
+	"/* This file allows Globals.h to be included with an absolute path */\n#include \"${PROJECT_SOURCE_DIR}/src/Globals.h\"\n")
 
 configure_file("BuildInfo.h.cmake" "${CMAKE_BINARY_DIR}/include/BuildInfo.h")
+target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE "${CMAKE_BINARY_DIR}/include/")
+
+# Enable PCH
+# TODO: upgrade CMake version and remove guard:
+if (PRECOMPILE_HEADERS AND CMAKE_VERSION VERSION_GREATER_EQUAL "3.16")
+	target_precompile_headers(${CMAKE_PROJECT_NAME} PRIVATE Globals.h)
+else()
+	message(WARNING "Precompiled headers for FASTER BUILDS not enabled, upgrade to CMake 1.16 or newer!")
+endif()
 
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+# Link process information library:
+if (WIN32)
+	target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE Psapi.lib)
+endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
 	set_source_files_properties(BlockType.cpp PROPERTIES COMPILE_FLAGS "-Wno-error=global-constructors")
 	set_source_files_properties(ByteBuffer.cpp PROPERTIES COMPILE_FLAGS "-Wno-error=global-constructors")
 	set_source_files_properties(ClientHandle.cpp PROPERTIES COMPILE_FLAGS "-Wno-error=global-constructors ")
 	set_source_files_properties(Statistics.cpp PROPERTIES COMPILE_FLAGS "-Wno-error=global-constructors")
 endif()
 
-if (NOT MSVC)
-	# Bindings need to reference other folders, so they are done here instead
-	# lib dependencies are not included
-	include_directories ("${CMAKE_CURRENT_SOURCE_DIR}/../lib/mbedtls/include")
-
-	foreach(folder ${FOLDERS})
-		add_subdirectory(${folder})
-	endforeach(folder)
-
-	get_directory_property(BINDING_DEPENDENCIES DIRECTORY "Bindings" DEFINITION BINDING_DEPENDENCIES)
-
-	#clear file
-	file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/Bindings/BindingDependencies.txt)
-	foreach(dependency ${BINDING_DEPENDENCIES})
-		#write each dependency on a seperate line
-		file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/Bindings/BindingDependencies.txt "${dependency}\n")
-	endforeach()
-
-	set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "Bindings.cpp Bindings.h")
```

Comment:
This seems like a nice idea, shame to drop it. Or does CMake clean generated output files automatically now?

Reply:
Yep :D Docs say the makefile generators automatically clean up and generated files, and can confirm during testing.